src/checkers/inference/solver/GeneralSolver.java

```diff
@@ -103,8 +103,40 @@ public InferenceSolution solve(Map<String, String> configuration, Collection<Slo
                     useGraph, solveInParallel);
             PrintUtils.writeStatistic(StatisticRecorder.getStatistic(), modelRecord, backEndType,
                     useGraph, solveInParallel);
+            postStatistic(constraints, slots, inferResult);
         }
-        return solution;
+
+        postVerification(constraints, slots, inferResult, qualHierarchy);
+
+        return new DefaultInferenceSolution(inferResult);
+    }
+
+    /**
+     * Process post-statistics on constraints, slots, and inference result after solver gives a solution.
+     *
+     * Sub-classes may override this method to perform customized statistics.
+     *
+     * @param constraints List of Constraints to be satisfied
+     * @param slots List of solts used in inference
+     * @param inferResult Map from slot id to inference solution for that slot
+     */
+    protected void postStatistic(final Collection<Constraint> constraints, final Collection<Slot> slots,
+            final Map<Integer, AnnotationMirror> inferResult) {
+        // Intentionally empty.
+    }
+
+    /**
+     * Process post-verification on the inference result.
+     *
+     * Sub-classes may override this method to perform customized verifications.
+     * @param constraints List of Constraints to be satisfied
+     * @param slots List of solts used in inference
+     * @param inferResult Map from slot id to inference solution for that slot
+     * @param qualifierHierarchy Target QualifierHierarchy
+     */
+    protected void postVerification(final Collection<Constraint> constraints, final Collection<Slot> slots,
+            final Map<Integer, AnnotationMirror> inferResult, final QualifierHierarchy qualifierHierarchy) {
+        // Intentionally empty.
```

Comment:
What example would you expected to be provided in here? 

Also, I'm thinking if it worth to have a general verification logic here, i.e. verify if solution satisfies "easy-checking" constraints (e.g. subtypeConstraint, equalityConstraint...).

Any suggestions?